File: Backend/src/InsureEdge.Domain/Entities/ClaimLetter.cs
namespace InsureEdge.Domain.Entities;

public class ClaimLetter
{
    public long Id { get; set; }
    public long ClaimId { get; set; }
    public long ClientId { get; set; }
    public string LetterCode { get; set; } = string.Empty;   // e.g. "LTR-0001"
    public string? LetterType { get; set; }
    public DateTimeOffset? LetterDate { get; set; }
    public DateTimeOffset? SendDate { get; set; }
    public string? Subject { get; set; }
    public string? LetterBody { get; set; }
    public string? RecipientRole { get; set; }
    public string? RecipientName { get; set; }
    public string? DeliveryMethod { get; set; }
    public string? RecipientEmail { get; set; }
    public string? RecipientAddress { get; set; }
    public string? Priority { get; set; }
    public bool FollowUp { get; set; }
    public string Status { get; set; } = "Draft";            // Draft | Sent
    public long? CreatedBy { get; set; }
    public DateTimeOffset CreatedOn { get; set; } = DateTimeOffset.UtcNow;
    public long? UpdatedBy { get; set; }
    public DateTimeOffset? UpdatedOn { get; set; }

    public User? CreatedByUser { get; set; }
    public User? UpdatedByUser { get; set; }
}

File: Frontend/src/api/claimLetter.ts
import api from './client';

export type ClaimLetterListItemDto = {
  id:              number;
  letterCode:      string;
  letterType:      string | null;
  letterDate:      string | null;
  sendDate:        string | null;
  recipientName:   string | null;
  recipientRole:   string | null;
  recipientEmail:  string | null;
  priority:        string | null;
  status:          string;
  createdByName:   string | null;
  createdOn:       string | null;
};

export type ClaimLetterDetailDto = {
  id:               number;
  letterCode:       string;
  letterType:       string | null;
  letterDate:       string | null;
  subject:          string | null;
  letterBody:       string | null;
  recipientRole:    string | null;
  recipientName:    string | null;
  deliveryMethod:   string | null;
  recipientEmail:   string | null;
  recipientAddress: string | null;
  priority:         string | null;
  followUp:         boolean;
  status:           string;
  sendDate:         string | null;
  createdByName:    string | null;
  createdOn:        string | null;
  updatedByName:    string | null;
  updatedOn:        string | null;
};

export type SaveClaimLetterRequest = {
  id:               number | null;
  claimId:          number;
  letterType:       string | null;
  letterDate:       string | null;
  subject:          string | null;
  letterBody:       string | null;
  recipientRole:    string | null;
  recipientName:    string | null;
  deliveryMethod:   string | null;
  recipientEmail:   string | null;
  recipientAddress: string | null;
  priority:         string | null;
  followUp:         boolean;
};

export const claimLetterApi = {
  getList: (claimId: number) =>
    api.get<ClaimLetterListItemDto[]>(`/claims/${claimId}/letters`).then(r => r.data),

  getById: (claimId: number, letterId: number) =>
    api.get<ClaimLetterDetailDto>(`/claims/${claimId}/letters/${letterId}`).then(r => r.data),

  save: (claimId: number, req: SaveClaimLetterRequest) =>
    api.post<{ id: number }>(`/claims/${claimId}/letters`, req).then(r => r.data),

  update: (claimId: number, letterId: number, req: SaveClaimLetterRequest) =>
    api.put<{ id: number }>(`/claims/${claimId}/letters/${letterId}`, req).then(r => r.data),

  send: (claimId: number, letterId: number) =>
    api.post<{ sent: boolean }>(`/claims/${claimId}/letters/${letterId}/send`).then(r => r.data),

  delete: (claimId: number, letterId: number) =>
    api.delete(`/claims/${claimId}/letters/${letterId}`).then(r => r.data),
};
